client/charts/js/components/HomepageSelection.jsx

```diff
@@ -39,7 +39,9 @@ export default function HomepageSelection({
 }) {
 	const years = d3
 		.groups(
-			originalDataset.map((d) => ({ year: d.date.getUTCFullYear() })),
+			originalDataset.map((d) => ({
+				year: (typeof d.date === "string" ? d3.timeParse("%Y-%m-%d")(d.date) : d.date).getUTCFullYear()
```

Comment:
Ran into this a few times while developing: Sometimes date comes in as a string formatted like `2023-01-02`, and sometimes it is an actual date object. There probably is a better way to do this instead of trying to detect a string and parsing it each time? Or should I just make this logic a function?

Reply:
Huh! I'm not seeing that locally (they're all strings for me) -- maybe something's different between our local databases?

Reply:
Oh, actually, it seems like they're supposed to be objects. Also weird!

Reply:
exactly, they're supposed to be objects but somehow are getting passed in as strings. Let me do some more digging to see if it makes more sense to do this parsing somewhere else.